public/posts/v1.6.md

```diff
@@ -0,0 +1,226 @@
+virtual file system that can be used to bundle assets. We are hoping that with
+future TC39 proposals like
+[`import assertions`](https://github.com/tc39/proposal-import-assertions), and
+[`asset references`](https://github.com/tc39/proposal-asset-references), the
+need for a virtual file system will disappear, because assets can be expressed
+right in the JS module graph.
+
+Currently the `deno compile` subcommand does not support cross platform
+compilation. Compilation for a specific platform has to happen on that platform.
+If there is demand, we would like to add the ability to cross compile for a
+different architechture using a `--target` flag when compiling. The tracking
+issue for this is [#8567](https://github.com/denoland/deno/issues/8567).
+
+Due to how the packaging of the binary works currently, a lot of unneeded code
+is included the binary. From preliminary tests we have determined that we could
+reduce the final binary size by around 60% (to around 20MB) when stripping out
+this unnecessary code. Work on this front is happening at the moment (e.g. in
+[#8640](https://github.com/denoland/deno/pull/8640)).
+
+## Built-in Deno Language Server
+
+<img src="/posts/v1.6/lsp.gif">
+
+Deno 1.6 ships with a new `deno lsp` subcommand that provides a language server
+implementing
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/).
+
+[VSCode Deno](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno)
+is the official VSCode extension that implements a lot of functionality by
+duplicating fundamental behaviours of Deno; a stark example is module resolution
+logic that requires module specifiers with extensions or support for import
+maps. Keeping extension logic in sync with Deno logic posed a significant
+challenge in itself, while providing high quality IDE experience in an editor
+other than VSCode would require to duplicate the efforts all over. Providing a
+self-contained LSP implementation in Deno makes it easier to provide a
+high-quality and unified editing experience for all code editors that understand
+LSP.
+
+The LSP is in the early stages of the development, but it already supports some
+crucial features:
+
+- Go to definition
+- Go to references
+- Hints on hover
+- `deno fmt` integration
+- `deno lint` integration
+
+To track the progress of the development follow
+[issue #8643](https://github.com/denoland/deno/issues/8643).
+
+[VSCode Deno Canary](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno-canary)
+is a new version of VSCode extension that uses `deno lsp`. It is also in the
+early stages of the development and is not yet as featureful as `VSCode Deno`.
+
+_If you decide to give it a try make sure to disable the old version of the
+extension, otherwise diagnostics might be duplicated. For now it is also
+recommended to use the Deno canary update channel when using the LSP, because
+updates and changes will still be frequent for the LSP._
```

Comment:
Does this sentence make sense? One can't use LSP outside of canary right?

Reply:
I mean the Deno CLI canary channel. I will clarify.